internal/containerupdate/container_updater.go

```diff
@@ -2,13 +2,71 @@ package containerupdate
 
 import (
 	"context"
+	"fmt"
 	"io"
 
+	"github.com/windmilleng/tilt/internal/container"
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/mode"
 	"github.com/windmilleng/tilt/internal/model"
 	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/target"
 )
 
 type ContainerUpdater interface {
 	UpdateContainer(ctx context.Context, deployInfo store.DeployInfo,
 		archiveToCopy io.Reader, filesToDelete []string, cmds []model.Cmd, hotReload bool) error
+	SupportsSpecs(specs []model.TargetSpec) (supported bool, msg string)
```

Comment:
The responsibility of the SupportsSpecs function seems poorly delineated. In other words, it seems like the decision about "am i allowed to live update in this container" is spread out across a few different helpers, and it's poorly defined which helpers are responsible for which parts of the decision